servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/P2CLoadBalancingPolicyBuilder.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.LoadBalancedConnection;
+
+import java.util.Random;
+import javax.annotation.Nullable;
+
+import static io.servicetalk.utils.internal.NumberUtils.ensurePositive;
+
+/**
+ * A builder for immutable {@link P2CLoadBalancingPolicy} instances.
+ */
+public final class P2CLoadBalancingPolicyBuilder {
```

Comment:
Similar to `AimdCapacityLimiterBuilder`, these top level builders should have a pkg-private default constructors. You can add `@see P2CLoadBalancingPolicyBuilder#p2c()` to help with discoverability.